tutorials/create-an-example-application.md

```diff
@@ -0,0 +1,444 @@
+			if err != nil {
+				log.Fatal(err)
+			}
+
+			// Create the database schema if it doesn't exist yet.
+			if _, err := db.Exec(schema); err != nil {
+				return err
+			}
+
+			// HTTP endpoint to provide basic Read/write operations according to
+			// a fixed SQL schema.
+			http.HandleFunc("/", func(w http.ResponseWriter, r *http.Request) {
+				key := strings.TrimLeft(r.URL.Path, "/")
+				result := ""
+				switch r.Method {
+				case "GET":
+					row := db.QueryRow(query, key)
+					if err := row.Scan(&result); err != nil {
+						result = fmt.Sprintf("Error: %s", err.Error())
+					}
+					break
+				case "PUT":
+					result = "done"
+					value, _ := io.ReadAll(r.Body)
+					if _, err := db.Exec(update, key, string(value[:])); err != nil {
+						result = fmt.Sprintf("Error: %s", err.Error())
+					}
+				default:
+					result = fmt.Sprintf("Error: unsupported method %q", r.Method)
+
+				}
+				fmt.Fprintf(w, "%s\n", result)
+			})
+
+			listener, err := net.Listen("tcp", api)
+			if err != nil {
+				return err
+			}
+
+			go http.Serve(listener, nil)
+
+			// wait until we received a termination signal
+			ch := make(chan os.Signal, 32)
+			signal.Notify(ch, unix.SIGPWR)
+			signal.Notify(ch, unix.SIGINT)
+			signal.Notify(ch, unix.SIGQUIT)
+			signal.Notify(ch, unix.SIGTERM)
+
+			<-ch
+
+			db.Close()
+			// Transfer all responsibilities (leader/voting rights) to other node
+			app.Handover(context.Background())
+			app.Close()
+
+			return nil
+		},
+	}
+
+	flags := cmd.Flags()
+	flags.StringVarP(&api, "api", "a", "", "address used to expose the demo API")
+	flags.StringVarP(&db, "db", "d", "", "address used for internal database replication")
+	join = flags.StringSliceP("join", "j", nil, "database addresses of existing nodes")
+	flags.StringVarP(&dir, "dir", "D", "/tmp/dqlite-demo", "data directory")
+	cmd.MarkFlagRequired("db")
+
+	if err := cmd.Execute(); err != nil {
+		os.Exit(1)
+	}
+}
+```
+
+Let's spin up three instances of the application to verify that everything works as expected:
+
+```bash
+go run go-dqlite-demo.go --api 127.0.0.1:8001 --db 127.0.0.1:9001 &
+go run go-dqlite-demo.go --api 127.0.0.1:8002 --db 127.0.0.1:9002 --join 127.0.0.1:9001 &
+go run go-dqlite-demo.go --api 127.0.0.1:8003 --db 127.0.0.1:9003 --join 127.0.0.1:9001 &
+```
+
+You should now be able to set:
+
+```bash
+ubuntu@dqlite-tutorial:~$ curl -X PUT -d my-value http://127.0.0.1:8001/my-key
+done
+```
+
+and retrieve values:
+```bash
+ubuntu@dqlite-tutorial:~$ curl http://127.0.0.1:8001/my-key
+my-value
+```
+
+The first node is currently the leader. Let's kill it to check if fail-over to other nodes works as expected:
+
+```bash
+ubuntu@dqlite-tutorial:~$ kill -TERM %1; curl http://127.0.0.1:8002/my-key
+my-value
+```
+
+As you can see, the leader node was killed but we can still retrieve the value from the other API servers.
+
+That's all for this tutorial. Check out the [Explanation](tbd) and [Reference](tbd) sections to find out more.
```

Comment:
```suggestion
That's all for this tutorial. Check out the Explanation and Reference sections to find out more.
```
We don't have links unfortunately - but the docs are small, so should be easy to find without. :)

Reply:
No need for all the empty lines at the end.